### ISSUE TITLE
good_bot Ignored?

### ISSUE THREAD

**Issue by cuemchugh** [mitchellkrogza/apache-ultimate-bad-bot-blocker#121]
A small number of users to our Apache 2.4 server use the Silk browser on their Kindle Fire tablets.  In the **globalblacklist.conf** Silk is among the bad_bot's listed.  I added the following to our **blacklist-user-agents.conf** but it doesn't appear to be working:
BrowserMatchNoCase "^(.*?)(\bSilk\b)(.*)$" good_bot

Unfortunately Silk users / bots are still getting 403'd:
"GET / HTTP/1.1" 403 3822 "https://www.bing.com/" "Mozilla/5.0 (Linux; Android 5.1.1; KFFOWI) AppleWebKit/537.36 (KHTML, like Gecko) Silk/72.4.5 like Chrome/72.0.3626.121 Safari/537.36"

The same appears to happen with any other user agent I make good_bot.

I applied the fix in [119 ](https://github.com/mitchellkrogza/apache-ultimate-bad-bot-blocker/issues/119#issue-413292247) & reloaded Apache so we're current but the issue persists.  Any ideas on how I can better troubleshoot why good_bot doesn't seem to be working for us?

**Comment by mitchellkrogza** [comment]
Thanks for reporting @cuemchugh at one point some big additions were introduced @davcpas1234  into the bad-user-agent.list and this is now the second false positive. 

I am going to bring this repo to match the list of bad user agents used on the Nginx Bot Blocker project and keep them in par with each other which I have always done. 

Any removals which will come in this commit (coming now) which users want to block can be added to their own blacklist-user-agents.conf include file.

**Comment by mitchellkrogza** [comment]
We will have to proceed with new additions to the bad user agents list on a case by case basis, templates for this will be added into the Issues logging system

**Comment by mitchellkrogza** [comment]
Fixed in https://github.com/mitchellkrogza/apache-ultimate-bad-bot-blocker/commit/5cd3e51081ad091617c29023f613aa5ef4afbcad

Reporting Template Added in https://github.com/mitchellkrogza/apache-ultimate-bad-bot-blocker/commit/4c9c2a5c3e5fba1c516e2670daf6704165003d11

**Comment by davcpas1234** [comment]
@cuemchugh I've had a look into this as I added Silk to the bad_user_agents list, this appears to be a conflict with a known Crawler.

```
silk/1.0 (+http://www.slider.com/silk.htm)/3.7
Silk/1.0
```

and the new Amazon Browser Silk

```
Mozilla/5.0 (Linux; Android android-version; product-model Build/product-build) AppleWebKit/webkit-version (KHTML, like Gecko) Silk/browser-version like Chrome/chrome-version Safari/webkit-version
```

CC: @mitchellkrogza

**Comment by cuemchugh** [comment]
@mitchellkrogza @davcpas1234 Thank you both: I updated our config this morning and saw Silk was removed.  It makes sense that Silk would be blocking given it's naming: I told the affected site that Silk can be a problematic name at this point.  I wish Amazon had thought of that much.

The thing I'm still confused by is good_bot not working?  I thought anything in the blacklist-user-agents.conf would supersede the global config?  I've added additional bad_bot entries to that config and that's been working fine so I wasn't sure if good_bot was working at this point.  Thanks again.
